server/src/main/java/org/candlepin/controller/CandlepinPoolManager.java

```diff
@@ -1547,71 +1551,71 @@ public void regenerateCertificatesOf(Owner owner, String productId, boolean lazy
-        Consumer consumer = entitlement.getConsumer();
-        Pool pool = entitlement.getPool();
-
-        // Similarly to when we add an entitlement, lock the pool when we remove one, too.
-        // This won't do anything for over/under consumption, but it will prevent
-        // concurrency issues if someone else is operating on the pool.
-        pool = poolCurator.lockAndLoad(pool);
-        consumer.removeEntitlement(entitlement);
+    public void revokeEntitlements(List<Entitlement> entsToRevoke) {
+        if (entsToRevoke.isEmpty()) {
+            return;
+        }
 
-        // Look for pools referencing this entitlement as their source
-        // entitlement and clean them up as well
+        List<Pool> poolsToDelete = poolCurator.listBySourceEntitlements(entsToRevoke);
 
-        // we need to create a list of pools and entitlements to delete,
-        // otherwise we are tampering with the loop iterator from inside
-        // the loop (#811581)
-        Set<Pool> deletablePools = new HashSet<Pool>();
+        List<Pool> poolsToLock = new ArrayList<Pool>();
+        for (Entitlement ent: entsToRevoke) {
+            poolsToLock.add(ent.getPool());
 
-        for (Pool p : poolCurator.listBySourceEntitlement(entitlement)) {
-            for (Entitlement e : p.getEntitlements()) {
-                this.revokeEntitlement(e);
+            // If we are deleting a developer entitlement, be sure to delete the
+            // associated pool as well.
+            if (ent.getPool() != null && ent.getPool().isDevelopmentPool()) {
+                poolsToDelete.add(ent.getPool());
             }
 
-            deletablePools.add(p);
-        }
-        for (Pool dp : deletablePools) {
-            deletePool(dp);
         }
 
-        pool.getEntitlements().remove(entitlement);
-        poolCurator.merge(pool);
-        entitlementCurator.delete(entitlement);
+        poolCurator.lock(poolsToLock);
+
+        log.info("Batch revoking {} entitlements ", entsToRevoke.size());
 
-        Event event = eventFactory.entitlementDeleted(entitlement);
-        if (!entitlement.isValid() &&
-                entitlement.getPool().isUnmappedGuestPool() &&
-                consumerCurator.getHost(consumer.getFact("virt.uuid"), consumer.getOwner()) == null) {
-            event = eventFactory.entitlementExpired(entitlement);
-            event.setMessageText(event.getMessageText() + ": " +
-                i18n.tr("Unmapped guest entitlement expired without establishing a host/guest mapping."));
+        entsToRevoke =  new ArrayList<Entitlement>(entsToRevoke);
+
+
+        for (Pool pool : poolsToDelete) {
+            entsToRevoke.addAll(pool.getEntitlements());
         }
 
-        // The quantity is calculated at fetch time. We update it here
-        // To reflect what we just removed from the db.
-        pool.setConsumed(pool.getConsumed() - entitlement.getQuantity());
+        poolCurator.lock(poolsToDelete);
```

Comment:
Do you think its better to make a single lock call instead of:
poolCurator.lock(poolsToLock); and poolCurator.lock(poolsToDelete); or am I missing something?